torch_geometric/data/feature_store.py

```diff
@@ -0,0 +1,354 @@
+        the attribute. This allows for :obj:`AttrView` to be indexed by
+        different values of attr_name. In particular, for a feature store that
+        has `feat` as an `attr_name`, the following code indexes into `feat`:
+
+        .. code-block:: python
+
+            store[group_name].feat[:]
+
+        """
+        if key in ['_attr', '_store']:
+            return super(AttrView, self).__getattribute__(key)
+
+        self._attr.attr_name = key
+        if self._attr.is_fully_specified():
+            return self._store.get_tensor(self._attr)
+        return self
+
+    def __setattr__(self, key, value):
+        r"""Supports attribute assignment to the backing :obj:`TensorAttr` of
+        an :obj:`AttrView`. This allows for :obj:`AttrView` objects to set
+        their backing attribute values. In particular, the following operation
+        sets the `index` of an :obj:`AttrView`:
+
+        .. code-block:: python
+
+            view = store.view(TensorAttr(group_name))
+            view.index = torch.Tensor([1, 2, 3])
+
+        """
+        if key in ['_attr', '_store']:
+            return super(AttrView, self).__setattr__(key, value)
+
+        TensorAttr.__setattr__(self._attr, key, value)
+
+    def __getitem__(
+        self,
+        index: IndexType,
+    ) -> Union['AttrView', FeatureTensorType]:
+        r"""Supports indexing the backing :obj:`TensorAttr` object by an
+        index or a slice. If the index operation results in a fully-specified
+        :obj:`AttrView`, a Tensor is returned. Otherwise, the :obj:`AttrView`
+        object is returned. The following operation returns a Tensor object
+        as a result of the index specification:
+
+        .. code-block:: python
+
+            store[group_name, attr_name][:]
+
+        """
+        self._attr.index = index
+        if self._attr.is_fully_specified():
+            return self._store.get_tensor(self._attr)
+        return self
+
+    def __call__(self) -> FeatureTensorType:
+        r"""Supports :obj:`AttrView` as a callable to force retrieval from
+        the currently specified attributes. In particular, this passes the
+        current :obj:`TensorAttr` object to a GET call, regardless of whether
+        all attributes have been specified. It returns the result of this
+        call. In particular, the following operation returns a Tensor by
+        performing a GET operation on the backing feature store:
+
+        .. code-block:: python
+
+            store[group_name, attr_name]()
+
+        """
+        return self._store.get_tensor(self._attr)
+
+    def __eq__(self, __o: object) -> bool:
+        r"""Compares two :obj:`AttrView` objects by checking equality of their
+        :obj:`FeatureStore` references and :obj:`TensorAttr` attributes."""
+        if not isinstance(__o, AttrView):
+            return False
+
+        return id(self._store) == id(__o._store) and self._attr == __o._attr
```

Comment:
Two feature stores may be equal even is there `id` is different. Maybe just implement a `__eq__` in `FeatureStore` as well?

Reply:
Sounds good, for the base class `__eq__` will just be an `id` comparison but subclasses can override.

Reply:
👍